### PR TITLE
Make offscreen canvas rendering context use offscreen canvas' size

### LINKED ISSUE #24465: Make offscreen canvas rendering context use offscreen canvas' size
Spec: https://html.spec.whatwg.org/multipage/canvas.html#dom-offscreencanvas-width

https://github.com/servo/servo/pull/24464 adds `width` and `height` members, but these are duplicated from the ones stored in the associated OffscreenCanvas. We should do the following:
* remove the duplicate members
* remove the Option from OffscreenCanvasRenderingContext2D's canvas member, since there is always an associated canvas
* use the associated canvas' size when necessary
* when OffscreenCanvas's SetWidth or SetHeight methods are called, send a CanvasMsg::Recreate message to the canvas paint thread (see CanvasRenderingContext2D::set_bitmap_dimensions for a model, called from HTMLCanvasElement::recreate_contexts)

@bblanke This would be a good task for your group to work on. It should allow some more tests to pass, too.

### PR DESCRIPTION
**Pull request by bblanke**
Updated the offscreen canvas rendering context to use the offscreen canvas' size. This involved upgrading several methods to accept u64 sizes.

Additionally, the code in OffscreenCanvas::SetWidth() and OffscreenCanvas::SetHeight() was updated to send CanvasMsg::Recreate to the canvas paint thread.

---
- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix #24465 

- [X] These changes do not require tests because the purpose of this was to tidy up code

### CODE REVIEW

**Review by jdm — CHANGES_REQUESTED** [review]
Looks great! I just have a couple pieces of cleanup to suggest.

**Comment by jdm** [comment]
```
  ▶ CRASH [expected OK] /2dcontext/compositing/2d.composite.canvas.source-over.html
  │ 
  │ 
  │ 
  │ 
  │ 
  │ assertion failed: Rect::from_size(size).contains_rect(&rect) (thread CanvasThread, at components/pixels/lib.rs:28)
  │ stack backtrace:
  │    0: servo::main::{{closure}}
  │    1: std::panicking::rust_panic_with_hook
  │              at src/libstd/panicking.rs:477
  │    2: std::panicking::begin_panic
  │    3: pixels::rgba8_get_rect
  │    4: canvas::canvas_data::CanvasData::read_pixels::{{closure}}
  │    5: canvas::azure_backend::<impl canvas::canvas_data::GenericDrawTarget for azure::azure_hl::DrawTarget>::snapshot_data
  │    6: canvas::canvas_data::CanvasData::read_pixels
  │    7: std::sys_common::backtrace::__rust_begin_short_backtrace
  │    8: __rust_maybe_catch_panic
  │              at src/libpanic_unwind/lib.rs:80
  │    9: core::ops::function::FnOnce::call_once{{vtable.shim}}
  │   10: <alloc::boxed::Box<F> as core::ops::function::FnOnce<A>>::call_once
  │              at /rustc/084beb83e0e87d673d5fabc844d28e8e8ae2ab4c/src/liballoc/boxed.rs:922
  │   11: <alloc::boxed::Box<F> as core::ops::function::FnOnce<A>>::call_once
  │              at /rustc/084beb83e0e87d673d5fabc844d28e8e8ae2ab4c/src/liballoc/boxed.rs:922
  │       std::sys_common::thread::start_thread
  │              at src/libstd/sys_common/thread.rs:13
  │       std::sys::unix::thread::Thread::new::thread_start
  │              at src/libstd/sys/unix/thread.rs:79
  │   12: start_thread
  │   13: clone

```

**Comment by jdm** [comment]
Looks like we'll want to be a bit more careful about our implementation of to_u64. Let's base it off of the same conversion methods from upstream: https://github.com/servo/euclid/blob/6dcc86f0e5370b5be45a955239131293bd336c66/src/rect.rs#L539-L546

File: components/canvas/canvas_data.rs [hunk 1]
```diff
@@ -1189,3 +1189,26 @@ impl RectToi32 for Rect<f64> {
...
+    fn to_u64(&self) -> Size2D<u64>;
+}
+
+impl Size2DExt for Size2D<f64> {
+    fn to_u64(&self) -> Size2D<u64> {
+        return Size2D::new(self.width as u64, self.height as u64);
+    }
+}
+
+pub trait RectExt_ {
```
[the remark this answers is not in the document]
Comment (bblanke) [reply components/canvas/canvas_data.rs]: I submitted PR #375!

**Pull request closed without merging**